Modules/Loadable/Segmentations/EditorEffects/qSlicerSegmentEditorAbstractEffect.h

```diff
@@ -243,6 +248,12 @@ public slots:
   /// NOTE: name must be defined in constructor in C++ effects, this can only be used in python scripted ones
   virtual void setName(QString name);
 
+  /// Get title of effect. Returns the effect's name when the title is empty
+  virtual QString title()const;
+  /// Set the title of the effect
+  /// NOTE: title must be defined in constructor in C++ effects, this can only be used in python scripted ones
+  virtual void setTitle(QString title);
```

Comment:
Why Python and C++ effects behave differently? We should keep the behavior the same for both (either both required to set the title in the constructor or neither).

Reply:
> Why Python and C++ effects behave differently?

I made this to be coherent with the current definition of the class in which `name` and `perSegment` properties are meant, in C++ effects, to only be initialized from the constructor, not through a setter.

The different behaviour is justified by the fact that the `setTitle` method is required for scripted effects to make the `title` property writable (even in the constructor). What is not the case for C++ effects in which initialization may be done in constructors without using the setter.

I will however do what you suggest and keep the same setter's behaviour for all effects types. It should not be a problem since that property is not as critical as the `name` which is used in other contexts.

Reply:
Yes, we want to keep the C++ and Python APIs as similar as possible to help people more easily go back and forth between languages.

Reply:
Thank you for your feedback, that's what is done in the last submitted commit.
`setTitle` is now only defined in `qSlicerSegmentEditorAbstractEffect` base class and inherited by all other derived effect classes.